src/components/TodosProvider/TodosProvider.tsx

```diff
@@ -0,0 +1,209 @@
+import { getTodos } from '../../api/todos';
+import * as todoService from '../../api/todos';
+import { errorMessages } from '../ErrorNotification';
+import { TodosContextType } from '../../types/contextTypes';
+import {
+  initialFilterStatus,
+  initialTodos,
+} from '../../types/initialContextValues';
+import { getFilteredTodos } from '../../utils/getFilteredTodos';
+
+interface Props {
+  children: React.ReactNode;
+}
+
+const TodosContext = createContext<TodosContextType>({
+  todos: initialTodos,
+  setTodos: () => {},
+  filterStatus: initialFilterStatus,
+  setFilterStatus: () => {},
+  filteredTodos: [],
+  isLoading: false,
+  errorMessage: '',
+  setErrorMessage: () => {},
+  isLoadingData: false,
+  setIsLoading: () => {},
+  selectedTodoIds: [],
+  setSelectedTodoIds: () => {},
+  tempTodo: null,
+  setTempTodo: () => {},
+  createNewTodo: () => {},
+  newTodoTitle: '',
+  setNewTodoTitle: () => {},
+  deleteTodo: () => {},
+  clearCompletedTodos: () => {},
+  updateTodo: () => {},
+});
+
+export const useTodos = () => useContext(TodosContext);
+
+export const TodosProvider: FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+
+  const [selectedTodoIds, setSelectedTodoIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+
+  const [filterStatus, setFilterStatus] =
+    useState<FilterStatus>(initialFilterStatus);
+
+  const [isLoadingData, setIsLoadingData] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const filteredTodos = getFilteredTodos(todos, filterStatus);
+
+  useEffect(() => {
+    setIsLoadingData(true);
+
+    getTodos()
+      .then(data => {
+        setTodos(data);
+      })
+      .catch(() => {
+        setErrorMessage(errorMessages.unableToLoadTodos);
+      })
+      .finally(() => setIsLoadingData(false));
+  }, []);
+
+  const createNewTodo = useCallback(
+    async (title: string) => {
+      try {
+        setIsLoading(true);
+        setSelectedTodoIds([...selectedTodoIds, todoService.USER_ID]);
+
+        setTempTodo({
+          id: todoService.USER_ID,
+          userId: todoService.USER_ID,
+          title,
+          completed: false,
+        });
+
+        const newTodo = await todoService.createTodo({
+          userId: todoService.USER_ID,
+          title,
+          completed: false,
+        });
+
+        setNewTodoTitle('');
+
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+      } catch (error) {
+        setSelectedTodoIds((ids: number[]) =>
+          ids.filter(id => id !== todoService.USER_ID),
+        );
+        setErrorMessage(errorMessages.unableToAddTodo);
+      } finally {
+        setIsLoading(false);
+        setSelectedTodoIds((ids: number[]) =>
+          ids.filter(id => id !== todoService.USER_ID),
+        );
```

Comment:
```suggestion
        setErrorMessage(errorMessages.unableToAddTodo);
      } finally {
        setSelectedTodoIds((ids: number[]) =>
          ids.filter(id => id !== todoService.USER_ID),
        );
```
You remove loaders in the finally block